src/app/downloads/downloads.component.ts

```diff
@@ -0,0 +1,194 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import { CastleService } from '../core';
+import { EpisodeModel, EpisodeMetricsModel, PodcastMetricsModel, INTERVAL_DAILY, FilterModel } from '../ngrx/model';
+import { castlePodcastMetrics, castleEpisodeMetrics, castleFilter } from '../ngrx/actions/castle.action.creator';
+
+@Component({
+  selector: 'metrics-downloads',
+  template: `
+    <prx-spinner *ngIf="isPodcastLoading || isEpisodeLoading"></prx-spinner>
+    <metrics-downloads-chart *ngIf="podcastMetrics && podcastMetrics.length > 0"></metrics-downloads-chart>
+    <p class="error" *ngIf="error">{{error}}</p>
+  `,
+  styleUrls: ['downloads.component.css']
+})
+export class DownloadsComponent implements OnInit, OnDestroy {
+  episodeStoreSub: Subscription;
+  allEpisodes: EpisodeModel[];
+  podcastMetricsStoreSub: Subscription;
+  podcastMetrics: PodcastMetricsModel[];
+  episodeMetricsStoreSub: Subscription;
+  episodeMetrics: EpisodeMetricsModel[];
+  filterStoreSub: Subscription;
+  filter: FilterModel;
+  isPodcastLoading = true;
+  isEpisodeLoading = true;
+  error: string;
+
+  constructor(private castle: CastleService, public store: Store<any>) {
+  }
+
+  ngOnInit() {
+    this.setDefaultFilter();
+
+    this.filterStoreSub = this.store.select('filter').subscribe((state: FilterModel) => {
+      let changedFilter = false;
+      if (this.isPodcastChanged(state)) {
+        this.filter.podcast = state.podcast;
+        changedFilter = true;
+
+        // we don't want to even look at these stores until we have the selected podcast
```

Comment:
To address your question: "In any case, wanted to ask why sometimes here the stores aren’t even created until we’re into the logic of another subscription?"
The reason is that these stores need to be filtered by the podcast id. The idea is that eventually this will give us the ability to cache data on the front end because the stores will hold more data than is being shown and will be filtered on the filter's podcast, episode list, interval and date range. I would like to put those filters in selectors rather than in components, just not there yet.

Reply:
👍